packages/ide-extension/src/panel/guidedAnswersPanel.ts

```diff
@@ -199,23 +198,18 @@ export class GuidedAnswersPanel {
      * @returns - search result including trees
      */
     private async getTrees(queryOptions: GuidedAnswersQueryOptions): Promise<GuidedAnswerTreeSearchResult> {
-        const showLoadingAnimation = queryOptions.paging?.offset === 0;
-        if (showLoadingAnimation) {
-            if (this.loadingTimeout) {
-                clearTimeout(this.loadingTimeout);
-            }
-            this.loadingTimeout = setTimeout(() => {
-                this.postActionToWebview(updateLoading(true));
-            }, 2000);
-        }
-        const trees = await this.guidedAnswerApi.getTrees(queryOptions);
-        if (this.loadingTimeout) {
-            clearTimeout(this.loadingTimeout);
+        const showLoader = queryOptions.paging?.offset === 0;
+        if (showLoader) {
```

Comment:
@oliversberzs, by removing the timeout logic here, we get a regression and we have the issue that was fixed with https://github.com/SAP/guided-answers-extension/pull/437 again.

You can easily reproduce this by slowly typing into the search filed, e.g. search for s - a - p. You will immediately recognize that the 'a' is blocked (of course, depending on your speed of typing).

Reply:
seems a bit wrong to me that we are both not showing the loader when a network call is actually loading and also blocking the user from typing after an arbitrary time.

shouldn't we instead solve this problem in the regular way of:
1. debouncing the input, so that the network does not happen after every keystroke
2. not blocking the user from typing
3. making the debounce only start happening after the first initial keystroke, so that our first result appears just as fast, but the user can keep typing if they want a different one

Reply:
Hi @oliversberzs, 
The current solution in main branch works as follows:

- after the first keystroke it waits for 100ms
- if there is another keystroke, timer is reset and it waits for another 100ms
- if no further keystroke appears within 100ms, search (network) is started
- when search is started, new timer (for loading animation) for 2s is started
- if the search results arrive within 2s, stop the timer
- if there is another search within 2s, cancel the timer and start a new one
- if there is no new search within 2s and the results have not arrived, show the loading animation

I remember a problem with the loading animation: as soon as it is shown, I can't focus the input field anymore. 

If you have a better solution please add your proposal as code and we can review.

Reply:
then I'll revert it to how it was before, so we can merge it this sprint